src/tasks/netcore/NetCoreTaskHelper.ts

```diff
@@ -0,0 +1,298 @@
+                }
+            }
+        ];
+    }
+
+    public async provideDockerRunTasks(context: DockerTaskScaffoldContext, options?: NetCoreTaskScaffoldingOptions): Promise<DockerRunTaskDefinition[]> {
+        options = options || {};
+        options.appProject = options.appProject || await NetCoreTaskHelper.inferAppProject(context.folder); // This method internally checks the user-defined input first
+        options.platformOS = options.platformOS || 'Linux';
+
+        return [
+            {
+                type: 'docker-run',
+                label: 'docker-run: debug',
+                dependsOn: ['docker-build: debug'],
+                dockerRun: {
+                    entrypoint: options.platformOS === 'Windows' ? 'ping' : 'tail',
+                    command: options.platformOS === 'Windows' ? '-t localhost' : '-f /dev/null',
+                    os: options.platformOS === 'Windows' ? 'Windows' : undefined, // Default is Linux so we'll leave it undefined for brevity
+                },
+                netCore: {
+                    appProject: unresolveWorkspaceFolderPath(context.folder, options.appProject)
+                }
+            }
+        ];
+    }
+
+    public async resolveDockerBuildOptions(context: DockerBuildTaskContext, buildDefinition: NetCoreBuildTaskDefinition): Promise<DockerBuildOptions> {
+        const buildOptions = buildDefinition.dockerBuild;
+        const helperOptions = buildDefinition.netCore || {};
+
+        helperOptions.appProject = await NetCoreTaskHelper.inferAppProject(context.folder, helperOptions); // This method internally checks the user-defined input first
+
+        // tslint:disable: no-invalid-template-strings
+        buildOptions.context = buildOptions.context || '${workspaceFolder}';
+        buildOptions.dockerfile = buildOptions.dockerfile || path.join('${workspaceFolder}', 'Dockerfile');
+        // tslint:enable: no-invalid-template-strings
+        buildOptions.tag = buildOptions.tag || await NetCoreTaskHelper.getImageName(helperOptions.appProject);
+        buildOptions.labels = buildOptions.labels || NetCoreTaskHelper.defaultLabels;
+
+        return buildOptions;
+    }
+
+    public async resolveDockerRunOptions(context: DockerRunTaskContext, runDefinition: NetCoreRunTaskDefinition): Promise<DockerRunOptions> {
+        const runOptions = runDefinition.dockerRun;
+        const helperOptions = runDefinition.netCore || {};
+
+        helperOptions.appProject = await NetCoreTaskHelper.inferAppProject(context.folder, helperOptions); // This method internally checks the user-defined input first
+
+        runOptions.containerName = runOptions.containerName || await NetCoreTaskHelper.getContainerName(helperOptions.appProject);
+        runOptions.labels = runOptions.labels || NetCoreTaskHelper.defaultLabels;
+        runOptions.os = runOptions.os || 'Linux';
+        runOptions.image = runOptions.image || await NetCoreTaskHelper.getImageName(helperOptions.appProject);

```

Comment:
>getImageName [](start = 71, length = 12)

Just want to verify: if in my build task I specify a full image name (e.g. myapp:v1), and in my run task I do not specify any image name, then the run will still figure out that the image name is myapp:v1, right?

Reply:
I'd intended to implement as such but I either forgot or it got removed at some point. Putting that in now. @philliphoff I'm adding it for Node as well.